lib/private_key.ml

```diff
@@ -81,25 +64,23 @@ module Asn = struct
          (optional ~label:"attributes" @@ implicit 0 (SET of Attributes)
          which are defined in X.501; but nobody seems to use them anyways *)
 
-  let private_of_cstruct ~sloppy =
-    Asn_grammars.decode (Asn.codec Asn.der (private_key_info ~sloppy))
+  let private_of_cstruct = rsa_private_of_cstruct
 
-  let private_to_cstruct =
-    snd (Asn_grammars.projections_of Asn.der (private_key_info ~sloppy:false))
+  let private_to_cstruct = rsa_private_to_cstruct
```

Comment:
this is not entirely right, instead:
```OCaml
  let (private_of_cstruct, private_to_cstruct) =
    projections_of Asn.der private_key_info
```

(the PKCS8 private key includes the algorithm and version (as defined above in private_key_info), while rsa_private does not)

Reply:
Woops, missed the diference, it should be fine now.